### PR TITLE
Jlc/add cms support for admin and course creator

### PR DESCRIPTION
**Pull request by johanv26**
# Description
This is related to the requirement of Nelp that is creating users but need to give course creator permissions.
The problem is that until the user login, the user will not appear in the course-creator table.
https://github.com/eduNEXT/edx-platform/blob/master/cms/djangoapps/contentstore/views/course.py#L1889-L1893
So this PR adds the option to add course creator directly from the admin model matching with the username of the user model.
**Development**
This is a remake of https://github.com/eduNEXT/edx-platform/pull/269 using the `eox-nelp`plugin.
**Configuration** 
Remember to configure you have to update your `cms/env/devstack.py` or your custom settings.

`FEATURES['ENABLE_CREATOR_GROUP'] = True`.
**Note:** Is already in stage environment to check.
https://studio.stage.nelp.gov.sa/admin/course_creators/coursecreator/add/
### Jira story
https://edunext.atlassian.net/browse/NELP-195?atlOrigin=eyJpIjoiNmFmOTc4YTk4NDczNGZiZGEwZjA2YzY5MjdlYzljNjQiLCJwIjoiaiJ9

### CODE REVIEW

File: eox_nelp/edxapp_wrapper/backends/course_creators_l_v1.py [hunk 1]
```diff
@@ -0,0 +1,23 @@
+"""Backend for course_creators module.
```
Comment (andrey-canon) [inline eox_nelp/edxapp_wrapper/backends/course_creators_l_v1.py:1]: the convention is `file_name_{first letter of the version}_v1.py ` so for this case this should be `course_creators_k_v1.py`

File: eox_nelp/admin/course_creators.py [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+    add_fieldsets = (
+        (None, {
+            'fields': ['username', 'state', 'state_changed', 'note', 'all_organizations', 'organizations']
+        }),
+    )
+
+    def has_add_permission(self, request):
+        return True
+    def has_delete_permission(self, request, obj=None):
+        return True
```
Comment (andrey-canon) [inline eox_nelp/admin/course_creators.py:33]:
```suggestion
    def has_add_permission(self, request):
        return True

    def has_delete_permission(self, request, obj=None):
        return True
```

File: eox_nelp/admin/course_creators.py [hunk 3]
```diff
@@ -0,0 +1,36 @@
+"""courseCreators admin file.
+Contains all the nelped admin models for course_creators.
+classes:
+    nelpCourseCreatorAdmin: EoxNelp CourseCreators admin class.
+"""
+from __future__ import absolute_import
```
Comment (andrey-canon) [inline eox_nelp/admin/course_creators.py:6]: generally we use this in applications that has support for  python 2 and  3, this is just for python 3 so do you need this for something else ?

File: eox_nelp/edxapp_wrapper/course_creators.py [hunk 4]
```diff
@@ -0,0 +1,15 @@
+"""Wrapper course_creator module file.
+This contains all the required dependencies from course_creators.
+Attributes:
+    backend:Imported ccx module by using the plugin settings.
+    CourseCreator: Wrapper courseCreator model.
```
Comment (andrey-canon) [inline eox_nelp/edxapp_wrapper/course_creators.py:5]:
```suggestion
    CourseCreator: Wrapper CourseCreator model.
```

File: eox_nelp/edxapp_wrapper/course_creators.py [hunk 5]
```diff
@@ -0,0 +1,15 @@
+"""Wrapper course_creator module file.
+This contains all the required dependencies from course_creators.
+Attributes:
+    backend:Imported ccx module by using the plugin settings.
```
Comment (andrey-canon) [inline eox_nelp/edxapp_wrapper/course_creators.py:4]: ccx?

File: eox_nelp/apps.py [hunk 6]
```diff
@@ -33,9 +28,27 @@ class eoxNelpConfig(AppConfig):
...
                 'devstack': {'relative_path': 'settings.devstack'},
             },
+        }
+    }
+
+
+class EoxNelpCMSConfig(AppConfig):
+    """App configuration"""
+    name = 'eox_nelp'
+    verbose_name = "eduNEXT Openedx Extensions"
```
Comment (andrey-canon) [inline eox_nelp/apps.py:38]: I'm not sure about this

File: eox_nelp/admin/course_creators.py [hunk 7]
```diff
@@ -0,0 +1,36 @@
...
+
+from eox_nelp.admin.register_admin_model import register_admin_model as register
+from eox_nelp.edxapp_wrapper.course_creators import (
+    CourseCreator,
+    CourseCreatorAdmin,
+)
+
+
+class NelpCourseCreatorAdmin(CourseCreatorAdmin):
+    """EoxSupport CertificateTemplate admin class.
```
Comment (andrey-canon) [inline eox_nelp/admin/course_creators.py:18]: EoxSupport? CertificateTemplate ?

File: eox_nelp/admin/course_creators.py [hunk 8]
```diff
@@ -0,0 +1,36 @@
+"""courseCreators admin file.
```
Comment (andrey-canon) [inline eox_nelp/admin/course_creators.py:1]:
```suggestion
"""CourseCreators admin file.
```

File: eox_nelp/admin/course_creators.py [hunk 9]
```diff
@@ -0,0 +1,36 @@
...
+Contains all the nelped admin models for course_creators.
+classes:
+    nelpCourseCreatorAdmin: EoxNelp CourseCreators admin class.
```
Comment (andrey-canon) [inline eox_nelp/admin/course_creators.py:4]:
```suggestion
    NelpCourseCreatorAdmin: EoxNelp CourseCreators admin class.
```

**Review by andrey-canon — CHANGES_REQUESTED** [review]

File: eox_nelp/admin/course_creators.py (hunk 3 above)
Comment (johanv26) [reply eox_nelp/admin/course_creators.py:6]: No, I think no cause we are using python3, I will remove it.

File: eox_nelp/apps.py (hunk 6 above)
Comment (johanv26) [reply eox_nelp/apps.py:38]: @andrey-canon what is the problem? THe name???

File: eox_nelp/edxapp_wrapper/backends/course_creators_l_v1.py (hunk 1 above)
Comment (johanv26) [reply eox_nelp/edxapp_wrapper/backends/course_creators_l_v1.py:1]: Actually, I put lilac cause is like a reference. Most of the eox-core backends don't have the koa version, and instead, use the lilac. That is the reason I set Lilac.
image

File: eox_nelp/apps.py (hunk 6 above)
Comment (andrey-canon) [reply eox_nelp/apps.py:38]: this could be Nelp Openedx Extension since "eduNEXT Openedx Extensions" is eox-core

Comment (johanv26) [reply eox_nelp/apps.py:38]: Ah ok heheh let me I change it.

File: eox_nelp/edxapp_wrapper/backends/course_creators_l_v1.py (hunk 1 above)
Comment (andrey-canon) [reply eox_nelp/edxapp_wrapper/backends/course_creators_l_v1.py:1]: They didn't test eox-core with koa so I think that is the reason. Anyway you can not ensure that this backend will work in lilac therefore you should set the name with the version that you use

Comment (johanv26) [reply eox_nelp/edxapp_wrapper/backends/course_creators_l_v1.py:1]: OK, you have a good point. Let me change it.

File: eox_nelp/admin/course_creators.py (hunk 2 above)
Comment (johanv26) [reply eox_nelp/admin/course_creators.py]: Resolved in https://github.com/eduNEXT/eox-nelp/pull/2/commits/a125a7569bbfd536f28897a1a2c94aa41e39a324

File: eox_nelp/admin/course_creators.py (hunk 3 above)
Comment (johanv26) [reply eox_nelp/admin/course_creators.py]: Resolved in https://github.com/eduNEXT/eox-nelp/pull/2/commits/a125a7569bbfd536f28897a1a2c94aa41e39a324

File: eox_nelp/edxapp_wrapper/course_creators.py (hunk 4 above)
Comment (johanv26) [reply eox_nelp/edxapp_wrapper/course_creators.py]: Resolved in https://github.com/eduNEXT/eox-nelp/pull/2/commits/a125a7569bbfd536f28897a1a2c94aa41e39a324

File: eox_nelp/edxapp_wrapper/course_creators.py (hunk 5 above)
Comment (johanv26) [reply eox_nelp/edxapp_wrapper/course_creators.py]: Resolved in https://github.com/eduNEXT/eox-nelp/pull/2/commits/a125a7569bbfd536f28897a1a2c94aa41e39a324

File: eox_nelp/apps.py (hunk 6 above)
Comment (johanv26) [reply eox_nelp/apps.py]: Resolved in https://github.com/eduNEXT/eox-nelp/pull/2/commits/a125a7569bbfd536f28897a1a2c94aa41e39a324

File: eox_nelp/admin/course_creators.py (hunk 7 above)
Comment (johanv26) [reply eox_nelp/admin/course_creators.py]: Resolved in https://github.com/eduNEXT/eox-nelp/pull/2/commits/a125a7569bbfd536f28897a1a2c94aa41e39a324

File: eox_nelp/admin/course_creators.py (hunk 8 above)
Comment (johanv26) [reply eox_nelp/admin/course_creators.py]: Resolved in https://github.com/eduNEXT/eox-nelp/pull/2/commits/a125a7569bbfd536f28897a1a2c94aa41e39a324

File: eox_nelp/admin/course_creators.py (hunk 9 above)
Comment (johanv26) [reply eox_nelp/admin/course_creators.py]: Resolved in https://github.com/eduNEXT/eox-nelp/pull/2/commits/a125a7569bbfd536f28897a1a2c94aa41e39a324

File: eox_nelp/edxapp_wrapper/backends/course_creators_l_v1.py (hunk 1 above)
Comment (johanv26) [reply eox_nelp/edxapp_wrapper/backends/course_creators_l_v1.py]: Changed in  71e8c18e9cd5977ac38a022e1f2387d45ab1d9a1

**Review by andrey-canon — APPROVED** [review]

**Pull request merged**